cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java

```diff
@@ -207,15 +195,16 @@ private void runGeneratedExecutableWithSameClassLoader(BLangPackage executableMo
         }
     }
 
-    private static String getClassPath(ExecutableJar jar) {
-        String seperator = ":";
-        if (OsUtils.isWindows()) {
-            seperator = ";";
-        }
-        StringBuilder classPath = new StringBuilder(jar.moduleJar.toString());
-        for (Path path : jar.platformLibs) {
-            classPath.append(seperator).append(path);
-        }
-        return classPath.toString();
+    private String getAllClassPaths(BLangPackage executableModule, BuildContext buildContext) {
+        // Since classpath separator depends on the OS type
+        String separator = OsUtils.isWindows() ? ";" : ":";
+        StringBuilder cp = new StringBuilder();
```

Comment:
You can use `StringJoiner` with `File.pathSeparator` here.

`StringJoiner joiner = new StringJoiner(File.pathSeparator);`

Reply:
@rasika That's a good suggestion! I did a rough timing analysis for both approaches and `StringJoiner` is slighly faster than StringBuilder in here. Thanks for pointing that out :)